### DISCUSSION TITLE
your context manager

### DISCUSSION THREAD

**Discussion by devanshkv** [thepetabyteproject/your#65] [Ideas]
All files should be closed after reading. `psrfits` does that to some extent but for `filterbanks` it needs to be `self.fp.close()`.

Those who do not know about the inner workings wouldn't know how to close files.
Should we introduce a context manager for `your`? the idea would be to do something like:

```python
import your

with your.Your("your_files") as your_object:
    your_object.do_stuff()

```
implementation is fairly easy, we can create `__enter__` and `__exit__` in `your.Your` as follows:
```python
def __enter__(self):
    # no need to do anything here
    return self

def __exit__(self, exc_type, exc_value, exc_traceback):
    # we implement .close_files() in all format classes
    return self.formatclass.close_files()
```

**Comment by KshitijAggarwal** [comment]
I think this would be useful. What changes would be required to the `Candidate` and `Writer` class? where `Your` is either being inherited or is saved as an attribute.
